src/app/common/code-block/ResourceCodeBlock.vue

```diff
@@ -90,25 +87,27 @@ const emit = defineEmits<{
   (event: 'reg-exp-mode-change', isRegExpMode: boolean): void
 }>()
 
-const yamlUniversal = computed(() => toYamlRepresentation(props.resource))
+const isCopying = ref(false)
+const text = ref('')
+const kCopyElement = ref<InstanceType<typeof KCopy> | null>(null)
 
-const noop: Copy = () => {}
+const yamlUniversal = computed(() => toYamlRepresentation(props.resource))
 
-const copy = ref<Copy>(noop)
-let p = new Promise((resolve: Resolve, reject) => { copy.value = (cb) => cb(resolve, reject) })
+// False negative. We should enable the TypeScript-specific linter rule, but that requires further changes.
+// eslint-disable-next-line no-extra-parens
+const copy = ref<(cb: CopyCallback) => void>(() => {})
+let promise = new Promise((resolve: Resolve, reject) => {
+  copy.value = (cb) => cb(resolve, reject)
+})
 
 const fetcher = async (_params?: SingleResourceParameters): Promise<Entity> => {
-  let res: Entity
   try {
-    res = await p
+    return await promise
```

Comment:
nit: Do we need to await the promise here if we are just returning it from an async function?

Reply:
Technically no. I removed it while rebasing the PR.